src/lib.rs

```diff
@@ -1290,6 +1359,124 @@ mod tests {
         assert_eq!(tsps.best_score, 5.8496657269155685);
     }
 
+    #[rstest(
```

Comment:
Can this be moved to `test_utils.rs` along with other constants we use in tests ?

Reply:
@hrshdhgd move what specifically? these are the parameters for this parameterized test, not sure how/why exactly you'd move them to a different file

Reply:
> (discuss at hackathon today?)

Yes.

> not sure how/why exactly you'd move them to a different file

The lib.rs file is getting too large (already ~2000 lines). It'll be hard to maintain if it explodes (by this I mean scrolling up and down too often which we are already kind of doing). We'll need a new PR to break out code into different modules once search is dealt with.